### PR TITLE
Always show label about awaiting payments

### LINKED ISSUE #6780: [Order Details] Show a note "Awaiting payment by Cash on Delivery" for unpaid CoD orders
Image

Previously, we showed a descriptive label saying "Awaiting payment via Cash on delivery" in the `Paid $0` row. Now that has been removed as superfluous information, but the "Awaiting payment" message was useful to make it clear to the merchant that the order has not been paid for yet.

- [ ] Add the "Awaiting payment" message back in its own row.

Context: p91TBi-8P0-p2#comment-10421

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #6780

### Description

Improvements on the order details screen regarding payment status (check the screenshots)

### Testing instructions

* Create an order via the app
* Notice unpaid state matches the image below 
* Pay for the order
* Notice unpaid state matches the image below 
* Create an order via the web (how via normal flow of buying something in a store as a customer)
* Notice unpaid state matches the image below 
* Pay for the order
* Notice unpaid state matches the image below 

### Images/gif

To align with the iOS implementation:

#### Paid order without payment title
image

#### Paid order with payment title
image

#### Unpaid order without payment title
image

#### Unpaid order with payment title

image

### CODE REVIEW

File: WooCommerce/src/main/res/values-zh-rTW/strings.xml [hunk 1]
```diff
@@ -2034,7 +2033,7 @@ Language: zh_TW
     <string name="logging_in">正在登入</string>
     <string name="all">全部</string>
     <string name="notification_channel_general_title">一般</string>
-    <string name="app_name">WooCommerce</string>
+    <string name="app_name">orderdetail_payment_summary_onhold</string>
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/res/values-zh-rTW/strings.xml]: ❓ Is this change intentional?

**Review by AnirudhBhat — APPROVED** [review]
Thank you for working on this @kidinov! I've tested this and it works as expected. The code changes look good to me as well! I've left a minor question that you might want to take a look at.

File: WooCommerce/src/main/res/values-zh-rTW/strings.xml (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/res/values-zh-rTW/strings.xml]: Wow, no. Apparently refactoring of AS did this. Fixed here 21efee67e45c4593ece1033deab714b9429db8c0

**Comment by kidinov** [comment]
@joshheald if you confirm that the functionally (check screenshots) it's what aligned with your understanding, please click merge button (I'll be AFK next 2 weeks)

**Comment by malinajirka** [comment]
I added the do not merge label because I believe we can’t modify other strings.xml except of the main one. It would fail the production build. We’ll need to introduce a new key for orderdetail_payment_summary_onhold and revert the changes in the non-main strings.xml files. I'll update the code next week since Andrei is afk and ask you for re-review @AnirudhBhat if that's fine.

**Comment by AnirudhBhat** [comment]
>I added the do not merge label because I believe we can’t modify other strings.xml except of the main one. It would fail the production build

Oh, TIL! Thanks for catching this! 

>I'll update the code next week since Andrei is afk and ask you for re-review @AnirudhBhat if that's fine.

Sure thing! Thank you!

**Comment by malinajirka** [comment]
@AnirudhBhat I've reverted the string changes in the non-main strings.xml files and removed the "do not merge label". Could you please review the last two commits. Thank you!

**Comment by AnirudhBhat** [comment]
Thanks for the changes @malinajirka LGTM! :shipit:
